Sources/Experiences/Services/RoverObserver.swift

```diff
@@ -96,7 +100,30 @@ public class RoverObserver {
                 queue: nil,
                 using: { [weak self] notification in
                     self?.trackPollAnswered(userInfo: notification.userInfo)
-            })
+            }),
+            
+            // Auto Tagging
+            NotificationCenter.default.addObserver(
+                forName: ScreenViewController.blockTappedNotification,
+                object: nil,
+                queue: nil,
+                using: { [weak self] (notification) in
+                    self?.tagUser(notificationName: ScreenViewController.blockTappedNotification, userInfo: notification.userInfo)
+            }),
+            NotificationCenter.default.addObserver(
+                forName: ScreenViewController.pollAnsweredNotification,
+                object: nil,
+                queue: nil,
+                using: { [weak self] (notification) in
+                    self?.tagUser(notificationName: ScreenViewController.pollAnsweredNotification, userInfo: notification.userInfo)
+            }),
+            NotificationCenter.default.addObserver(
+                forName: ScreenViewController.screenPresentedNotification,
+                object: nil,
+                queue: nil,
+                using: { [weak self] (notification) in
+                    self?.tagUser(notificationName: ScreenViewController.screenPresentedNotification, userInfo: notification.userInfo)
+            }),
```

Comment:
Hmmm thinking about this I should it before the other observers run. I want to ensure we add the new tag to when a block is clicked right away rather than future events

Reply:
@orospakr do observers run sync or async?

Reply:
I don't think we can make guarantees about the execution order of observers in Notification Center.